src/twisted/internet/_win32serialport.py

```diff
@@ -56,40 +59,54 @@ def __init__(self, protocol, deviceNameOrPortNumber, reactor,
         self._finishPortSetup()
 
 
+    @property
+    def _serialHandle(self):
+        # Try to detect pyserial version to support internal API changes
+        if hasattr(self._serial, '_port_handle'):
```

Comment:
`hasattr` can mask unexpected exceptions.  I'd suggest something like the following here:

```
serial = self._serial
try:
    # >= 3.0
    return serial._port_handle 
except AttributeError:
    # <= 2.7
    return serial.hComPort
```